### DISCUSSION TITLE
Add more typescript-eslint rules

### DISCUSSION THREAD

**Discussion by onemen** [epicweb-dev/config#7] [ESLint Suggestion]
### Search Terms Used

typescript-eslint

### Rule Docs

https://typescript-eslint.io/users/configs/#recommended-configurations

### Suggested Change

Since all all epicweb-dev packages uses typescript, I thing you should consider use some of `typescript-eslint` 
[recommended-configurations](https://typescript-eslint.io/users/configs/#recommended-configurations)

currently you only use `@typescript-eslint/no-unused-vars` and `@typescript-eslint/consistent-type-imports`

For example, linting `epic-stack` with [strict-type-checked](https://typescript-eslint.io/users/configs#strict-type-checked) and [stylistic-type-checked](https://typescript-eslint.io/users/configs#stylistic-type-checked) result with 1014 errors.

I know that `typescript-eslint` strict configurations are opinionated but I would like to here your opinion on this.

### Issue Likelihood

`typescript-eslint` recommendation is to use one of its [recommended configurations](https://typescript-eslint.io/users/configs/#recommended-configurations)

### Issue Severity

same as above

### Additional Context

```shell
npx @eslint/config-inspector@latest
```
Using the above command in `epic-stack` folder show that the current eslint configuration using only **19** rules in total.

**Comment by kentcdodds** [comment]
Just because rules exist and the epic stack breaks them does not mean they protect from actual issues that surface in poor user experience. 

I would like to discuss each rule individually. Could you justify the inclusion of specific rules?

**Comment by onemen** [comment]
You are more experience, I would like to get your opinion.
It look odd to that you currently only using 2 rules from @typescript-eslint

**Reply by kentcdodds** [reply]
We only enable rules for which there is a justifiable reason: https://github.com/epicweb-dev/config/blob/main/docs/decisions/002-minimal-eslint.md

There are certainly some rules that I could/should enable, but I haven't taken the time to comb through all of them to determine which should be enabled. I'm definitely going to blindly enable everything some other team says is "recommended" though. We'll enable each rule on a case-by-case basis.

**Comment by onemen** [comment]
these are some rules that I think should be enabled for typescript in strict mode  

```
// promises rules
'@typescript-eslint/await-thenable'
'@typescript-eslint/no-floating-promises'
'@typescript-eslint/no-misused-promises'
'@typescript-eslint/prefer-promise-reject-errors'
'@typescript-eslint/require-await'

// be safe
'@typescript-eslint/no-unsafe-argument'
'@typescript-eslint/no-unsafe-assignment'
'@typescript-eslint/no-unsafe-call'
'@typescript-eslint/no-unsafe-member-access'
'@typescript-eslint/no-unsafe-return'

'@typescript-eslint/no-base-to-string' // prevnet '[object Object]'
'@typescript-eslint/no-non-null-assertion'
'@typescript-eslint/only-throw-error'
'@typescript-eslint/restrict-template-expressions'

'@typescript-eslint/no-confusing-void-expression'

```

**Reply by kentcdodds** [reply]
I've evaluated each of these and enabled some, but kept most disabled. However, I documented why for each case: https://github.com/epicweb-dev/config/commit/a08f68445e8161ef40f47d7291e25ccf460acf61

**Comment by kentcdodds** [comment]
Thank you for the suggestions!
